app/src/main/java/com/cmput301f20t21/bookfriends/repositories/BookRepository.java

```diff
@@ -42,37 +53,118 @@ public static BookRepository getInstance() {
         data.put("owner", owner);
         // when a book is first added, the status will always be "AVAILABLE"
         data.put("status", BOOK_STATUS.AVAILABLE.toString());
+
+        if (imageUriFile != null) {
+            return bookCollection.add(data).continueWithTask(task -> {
```

Comment:
could we simplify this process if we add image first and receive the url? Then we can add the book all at once.

Reply:
we can, but we do need the book id as the image key right? to do so we need to come up with a different key for images. But imo it does not simplify that much as we always have two requests to await

Reply:
good point, I forgot about the image key.